databricks/notebooks/rapids_airline_hyperopt.ipynb

```diff
@@ -0,0 +1,419 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Hyper-parameter Optimization with RAPIDS + MLflow + Hyperopt\n",
+    "\n",
+    "### Introduction\n",
+    "\n",
+    "&nbsp;&nbsp;&nbsp;&nbsp;Hyperparameter optimization is the task of picking the values for the hyperparameters of the model that provide the optimal results for the problem, as measured on a test dataset. This is often a crucial step and can help boost the model performance when done correctly. Despite its theoretical importance, HPO has been difficult to implement in practical applications because of the resources needed to run so many distinct training jobs. In this notebook, we explore the combination of RAPIDS, MLflow and Hyperopt to perform HPO on GPU and compare them with the performance of the CPU runs. We want to illustrate that HPO can be performed in an efficient manner with RAPIDS libraries on GPU.\n",
+    "      \n",
+    "### MLFlow\n",
+    "\n",
+    "&nbsp;&nbsp;&nbsp;&nbsp;[MLflow](https://mlflow.org/) is used for managing the machine learning lifecycle. It provides a way to track experiements and store data about them. MLflow also provides a way to deploy ML models. We'll make use of MLflow to store information about the models and use the built-in integration of MLflow into Databricks to register and store models.\n",
+    "\n",
+    "### Hyperopt\n",
+    "\n",
+    "[Hyperopt](http://hyperopt.github.io/hyperopt/) is a library for finding the best hyperparameters for a given objective function. It provides a way to choose the objective function, the search algorithm, the database in which to store the eval points. We'll use this to define the parameter space to search over. \n",
+    "\n",
+    "We'll use airlines data for this demo - the aim is to predict if an airline will be delayed or not using a target variable `ArrDelayBinary`. We'll use a Random Forest Classifier as the model for the learning task. \n",
```

Comment:
Probably "We'll use FAA flight history data for this demo - the aim is to predict if a given flight ..."